# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/Element.webidl
# Commit: 1954ce9669b8
# Full Hash: 1954ce9669b86b61c418c8286fbe0754ad2419cd
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 3e45e3d15eed -r 1954ce9669b8 dom/webidl/Element.webidl
--- a/dom/webidl/Element.webidl	Wed May 15 03:01:18 2024 +0000
+++ b/dom/webidl/Element.webidl	Wed May 15 03:01:19 2024 +0000
@@ -276,6 +276,8 @@
   SlotAssignmentMode slotAssignment = "named";
   [Pref="dom.webcomponents.shadowdom.declarative.enabled"]
   boolean clonable = false;
+  [Pref="dom.webcomponents.shadowdom.declarative.enabled"]
+  boolean serializable = false;
 };
 
 // https://dom.spec.whatwg.org/#element
@@ -403,8 +405,15 @@
   undefined setHTML(DOMString aInnerHTML, optional SetHTMLOptions options = {});
 };
 
+dictionary GetHTMLOptions {
+  boolean serializableShadowRoots = false;
+  sequence<ShadowRoot> shadowRoots = [];
+};
+
 partial interface Element {
   // https://html.spec.whatwg.org/#dom-element-sethtmlunsafe
   [Pref="dom.webcomponents.shadowdom.declarative.enabled"]
   undefined setHTMLUnsafe(DOMString html);
+  [Pref="dom.webcomponents.shadowdom.declarative.enabled"]
+  DOMString getHTML(optional GetHTMLOptions options = {});
 };